delphi/translators/for2py/f2grfn.py

```diff
@@ -217,33 +222,50 @@ def generate_python_src(
                 Script: <pyTranslate.py>"
         )
 
-        output_list = []
-        for item in python_source:
-            if item[2] == "module":
-                module_file = f"{temp_dir}/m_{item[1].lower()}.py"
-                try:
-                    with open(module_file, "w") as f:
-                        output_list.append("m_" + item[1].lower() + ".py")
-                        f.write(item[0])
-                except IOError:
-                    assert False, f"Unable to write to {module_file}."
-            else:
-                try:
-                    with open(python_file_name, "w") as f:
-                        output_list.append(python_file_name)
-                        f.write(item[0])
-                except IOError:
-                    assert False, f"Unable to write to {python_file_name}."
+    output_list = []
+    for item in python_source:
+        if item[2] == "module":
+            module_file_generator(item, temp_dir, output_list, python_files)            
+        else:
+            try:
+                with open(python_files[0], "w") as f:
+                    output_list.append(python_files[0])
+                    f.write(item[0])
+            except IOError:
+                assert False, f"Unable to write to {python_file_name}."
 
-        try:
-            with open(output_file, "w") as f:
-                for fileName in output_list:
-                    f.write(fileName + " ")
-        except IOError:
-            assert False, f"Unable to write to {output_file}."
+    try:
+        with open(output_file, "w") as f:
+            for fileName in output_list:
+                f.write(fileName + " ")
+    except IOError:
+        assert False, f"Unable to write to {output_file}."
 
     return python_source
 
+def module_file_generator(item, temp_dir, output_list, python_files):
+    """This function extracts a translated module from
+    the python source and generates a new separate python file.
+
+    Args:
+        item (list): A list that each element holds a translated
+        python source in string.
+        temp_dir (str): A path to the temporary directory that will
+        hold generated files.
+        output_list (list): A list that holds list of output files.
+        python_files (list): A list that holds the generated python
+        python file paths.
+    ReturnL
+        None
+    """
+    module_file = f"{temp_dir}/m_{item[1].lower()}.py"
+    try:
+        with open(module_file, "w") as f:
+            output_list.append("m_" + item[1].lower() + ".py")
```

Comment:
The file name construction is repeated in lines 261 and 264.  The file name should be constructed in exactly one place.

Reply:
@skdebray I've reflected your suggestion in the code by declaring`MODULE_FILE_PREFIX` variable that holds` `m_` string. Reset of the code will use this variable to form the module file name or reference to check whether it's a module file or not. We can simply modify this variable in the future if we wish to specify the module file name with another prefix before the module name.